improver/fire_weather/build_up_index.py

```diff
@@ -0,0 +1,78 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+import numpy as np
+from iris.cube import Cube
+
+from improver.fire_weather import FireWeatherIndexBase
+
+
+class BuildUpIndex(FireWeatherIndexBase):
+    """
+    Plugin to calculate the Build Up Index (BUI) following
+    the Canadian Forest Fire Weather Index System.
+
+    The BUI is a numerical rating of the total amount of fuel available
+    for combustion. It combines the Duff Moisture Code (DMC) and the
+    Drought Code (DC) to represent the fuel buildup.
+
+    This process is adapted directly from:
+        Equations and FORTRAN Program for the
+        Canadian Forest Fire Weather Index System
+        (C.E. Van Wagner and T.L. Pickett, 1985).
+        Page 7, Equations 27a-27b.
+
+    Expected input units:
+        - Duff Moisture Code (DMC): dimensionless
+        - Drought Code (DC): dimensionless
+    """
+
+    INPUT_CUBE_NAMES = ["duff_moisture_code", "drought_code"]
+    OUTPUT_CUBE_NAME = "build_up_index"
+    # Map input cube names to internal attribute names for consistency
+    INPUT_ATTRIBUTE_MAPPINGS = {
+        "duff_moisture_code": "input_dmc",
+        "drought_code": "input_dc",
+    }
+
+    input_dmc: Cube
+    input_dc: Cube
+
+    def _calculate(self) -> np.ndarray:
+        """Calculates the Build Up Index (BUI) from DMC and DC.
+
+        From Van Wagner and Pickett (1985), Page 7: Equations 27a-27b.
+
+        Returns:
+            np.ndarray: The calculated BUI values.
+        """
+        dmc_data = self.input_dmc.data
+        dc_data = self.input_dc.data
+
+        # Condition 1: If both DMC and DC are zero, set BUI = 0
+        both_zero = (dmc_data == 0.0) & (dc_data == 0.0)
+
+        # Condition 2: If DMC <= 0.4 * DC use equation 27a
+        use_27a = dmc_data <= 0.4 * dc_data
```

Comment:
The Fortran has `<` instead of `<=`. I doubt this will have any noticeable impact.